src/app.service.ts

```diff
@@ -0,0 +1,99 @@
+import { Context } from 'koa';
+import { redisGetJson, redisSetJson } from './clients/redis';
+import { LoggerService } from './helpers';
+import { ChannelResult } from './interfaces/channel-result';
+import { CustomerCreditTransferInitiation } from './interfaces/iPain001Transaction';
+import { Channel, NetworkMap } from './interfaces/network-map';
+import { RuleResult } from './interfaces/rule-result';
+import { TypologyResult } from './interfaces/typology-result';
+import apm from 'elastic-apm-node';
+
+export const handleChannels = async (
+  ctx: Context,
+  transaction: CustomerCreditTransferInitiation,
+  networkMap: NetworkMap,
+  ruleResult: RuleResult[],
+  typologyResult: TypologyResult,
+  channelResult: ChannelResult,
+  channel: Channel,
+): Promise<{
+  transactionID: string;
+  message: string;
+}> => {
+  try {
+    const transactionID =
+      transaction.PaymentInformation.CreditTransferTransactionInformation
+        .PaymentIdentification.EndToEndIdentification;
+
+    // Initialize the result message
+    const result = {
+      transactionID: transactionID,
+      message: '',
+    };
+
+    // Check if the channel is completed
+    const hasChannelCompleted = await checkChannelCompletion(
+      transactionID,
+      channel,
+      typologyResult,
+    );
+
+    // If the channel is completed, then save the trasaaction evaluation result
+    if (hasChannelCompleted) {
+      const transactionHistoryQuery = `
+      INSERT {
+        "transactionID": ${JSON.stringify(transactionID)},
+        "transaction": ${JSON.stringify(transaction)},
+        "networkMap": ${JSON.stringify(networkMap)},
+        "ruleResult": ${JSON.stringify(ruleResult)},
+        "typologyResult": ${JSON.stringify(typologyResult)},
+        "channelResult": ${JSON.stringify(channelResult)}
+    } INTO "history"
+    `;
+
+      await ctx.state.arangodb.query(transactionHistoryQuery);
+
+      result.message = 'The transaction evaluation result is saved.';
+    }
```

Comment:
I think you're missing an Else here - because the next line would always override the message in the IF

Reply:
Yeap - fixed! 👍